src/components/challenge-rewards-drawer/ReferralLinkCopyButton.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useContext, useCallback, useRef } from 'react'
+
+import Clipboard from '@react-native-clipboard/clipboard'
+import { getUserHandle } from 'audius-client/src/common/store/account/selectors'
+import { Animated, StyleSheet, View, TouchableHighlight } from 'react-native'
+import LinearGradient from 'react-native-linear-gradient'
+
+import IconCopy from 'app/assets/images/iconCopy.svg'
+import Text from 'app/components/text'
+import { useSelectorWeb } from 'app/hooks/useSelectorWeb'
+import { useThemedStyles } from 'app/hooks/useThemedStyles'
+import { ThemeColors, useThemeColors } from 'app/utils/theme'
+
+import { ToastContext } from '../toast/ToastContext'
+
+const messages = {
+  copyPrompt: 'Copy Invite Link',
+  copyNotice: 'Referral link copied to the clipboard'
+}
+const createStyles = (themeColors: ThemeColors) =>
+  StyleSheet.create({
+    copyPromptContainer: {
+      width: '100%'
+    },
+    borderRadius: {
+      borderRadius: 6
+    },
+    copyPromptContent: {
+      padding: 16
+    },
+    copyPrompt: {
+      marginBottom: 12,
+      fontSize: 16,
+      textTransform: 'uppercase',
+      textAlign: 'center',
+      color: '#c675ff',
+      letterSpacing: 1
+    },
+    copyText: {
+      display: 'flex',
+      flexDirection: 'row',
+      alignItems: 'center',
+      justifyContent: 'center'
+    },
+    referralUrl: {
+      fontSize: 16,
+      textAlign: 'center',
+      color: themeColors.staticWhite
+    },
+    iconCopy: {
+      color: themeColors.staticWhite,
+      lineHeight: 16,
+      marginLeft: 8
+    }
+  })
+
+export const ReferralLinkCopyButton = () => {
+  const {
+    pageHeaderGradientColor1,
+    pageHeaderGradientColor2
+  } = useThemeColors()
+  const styles = useThemedStyles(createStyles)
+
+  const handle = useSelectorWeb(getUserHandle)
+  const referralUrl = `audius.co/signup?ref=${handle}`
+
+  const { toast } = useContext(ToastContext)
+  const onCopyClicked = useCallback(() => {
+    Clipboard.setString(referralUrl)
+    toast({ content: messages.copyNotice, type: 'info' })
+  }, [referralUrl, toast])
+
+  // Button animation
+  const scale = useRef(new Animated.Value(1)).current
+  const handlePressIn = () => {
```

Comment:
Maybe we should generic/hookify this one

Reply:
#234 for either fast follow or to merge into this :)